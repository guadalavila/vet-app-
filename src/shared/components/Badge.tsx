import React from 'react';
import { StyleSheet, View } from 'react-native';
import { colors } from '../utils/colors';
import { size } from '../utils/size';
import CustomText from './CustomText';

interface IBadgeProps {
    label: string;
    color?: string;
}
const Badge = ({ label, color = colors.light.primary }: IBadgeProps) => {
    return (
        <View testID='container-text' style={[styles.container, { backgroundColor: color }]}>
            <CustomText style={styles.label}>{label}</CustomText>
        </View>
    );
};

export default Badge;

const styles = StyleSheet.create({
    container: {
        borderRadius: 12,
        margin: size.M,
        padding: size.M,
    },
    label: {
        fontWeight: '600',
        color: colors.light.blackSecondary,
    },
});
