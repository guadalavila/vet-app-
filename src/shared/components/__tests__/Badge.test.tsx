import { render, screen } from '@testing-library/react-native';
import React from 'react';
import Badge from '../Badge';

describe('Badge', () => {
    test('should match snapshot', () => {
        const banner = render(<Badge label='Hello world' />);
        expect(banner).toMatchSnapshot();
    });

    test('should render text correct', () => {
        render(<Badge label='Hello world' />);
        expect(screen.getByText('Hello world')).toBeTruthy();
    });

    test('should render text styles correct', () => {
        render(<Badge label='Hello world' />);
        expect(screen.getByText('Hello world').props.style).toEqual([
            { color: '#11100F', fontSize: 16 },
            { color: '#11100F', fontWeight: '600' },
        ]);
    });

    test('should render correct color', () => {
        render(<Badge label='Hello world' color='#007cad' />);
        expect(screen.getByTestId('container-text')).toBeTruthy();
        expect(screen.getByTestId('container-text').props.style).toEqual([
            { borderRadius: 12, margin: 6, padding: 6 },
            { backgroundColor: '#007cad' },
        ]);
    });
});
