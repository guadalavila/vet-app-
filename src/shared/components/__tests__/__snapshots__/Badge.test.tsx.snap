// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 12,
        "margin": 6,
        "padding": 6,
      },
      Object {
        "backgroundColor": "#A765C1",
      },
    ]
  }
  testID="container-text"
>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
        },
        Object {
          "color": "#11100F",
          "fontWeight": "600",
        },
      ]
    }
  >
    Hello world
  </Text>
</View>
`;
